File: 1678-goal-parser-interpretation/1678-goal-parser-interpretation.java
class Solution {
    public String interpret(String command) {
        StringBuilder s = new StringBuilder(command.length());
        for(int i=0; i<command.length();i++){
            if(command.charAt(i)=='G'){
                s.append("G");
            }else if(command.charAt(i)=='('){
                if(command.charAt(i+1)==')'){
                    s.append("o");
                }else if(command.charAt(i+1)=='a'){
                    s.append("al");
                }
            }
        }
        return s.toString();
    }
}

File: 1678-goal-parser-interpretation/1678-goal-parser-interpretation.js
/**
 * @param {string} command
 * @return {string}
 */

//P: always string, no empty string
//R: string
//E: G()()()()(al)=>Gooooal
var interpret = function(command) {
  let str=[];
 for(let i=0;i<command.length;i++){
   if(command[i]=="G")
     str.push("G");
   else if(command[i]=="(" && command[i+1]==")"){
    str.push("o");
    i+=1
   } 
   else if(command[i]=="(" && command[i+1]=="a"){
    str.push("al");
     i+=3;
   }
     
 }
 return str.join("");
};